src/main/java/net/sf/jabref/shared/DBMSType.java

```diff
@@ -76,13 +75,26 @@ public int getDefaultPort() {
     }
 
     public static Optional<DBMSType> fromString(String typeName) {
-        for (DBMSType dbmsType : EnumSet.allOf(DBMSType.class)) {
-            if (typeName.equals(dbmsType.toString())) {
-                return Optional.ofNullable(dbmsType);
-            }
+        try {
+            return Optional.of(Enum.valueOf(DBMSType.class, typeName.toUpperCase()));
+        } catch (IllegalArgumentException exception) {
+            return Optional.empty();
         }
-        return Optional.empty();
     }
 
+    /**
+     * Escapes parts of SQL expressions like table or field name to match the conventions
+     * of the current database system type.
+     * @param expression Table or field name
+     * @return Correctly escaped expression
+     */
+    public String escape(String expression) {
+        if (this == DBMSType.MYSQL) {
+            return "`" + expression + "`";
```

Comment:
Define the Escape-Char as private final String/Char somewhere, makes it easier to identify it and to possible change it.